### PR TITLE
Activity log: Add empty period message

### LINKED ISSUE #14662: Activity Log/Rewind: Need an empty state view
Upon successful JP connection and setup, if the user visits the Activity tab in Stats, they will see an empty state.

This will appear if the view has no activity recorded yet.

image

### PR DESCRIPTION
**Pull request by sirreal**
Adds an empty notice when there are no logs to display.

~Note that the empty component is essentially a stub right now, I'm hoping to get some design and or copy feedback. That could happen in this PR or after.~

Apologies, this PR _should_ be small, but I accidentally committed a prettier format (#17039) 🙁 

## Testing
1. Visit https://calypso.live/stats/activity?branch=add/activitylog-placeholder&startDate=2017-07-01
1. Ensure you're at an empty view (navigating a month back will work ATM).
1. You should see the notice.
1. Ensure that loading state does not show empty notice.
1. Ensure that views _with_ logs are unaffected.

## Screens

[updated capture](https://github.com/Automattic/wp-calypso/pull/17032#issuecomment-321246943)

no-activity

Fixes #14662

### CODE REVIEW

File: client/my-sites/stats/activity-log/activity-log-empty.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { localize } from 'i18n-calypso';
+
+function ActivityLogEmpty( { translate } ) {
+	return (
+		<div>
+			{ translate( 'No activity for this period' ) }
```
Comment (a8ci18n) [inline client/my-sites/stats/activity-log/activity-log-empty.js]: Hi! I've found a possible matching string that has already been translated **24** times:
`translate( 'No activity this period' )` *ES Score: 12*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/activitylog-placeholder)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by folletto** [comment]
> I'm hoping to get some design and or copy feedback. That could happen in this PR or after.

I'd use `EmptyContent` here too. :)

* Text: "No activity for this period" or even better "No activity for July 2017" (same frame as above)
* Subtitle: none
* Action: none
* Illustration: `illustration-empty-results.svg`

**Comment by sirreal** [comment]
Nice pointers, thanks @folletto! Pushed changes, see updated screenshot in the description.

File: client/my-sites/stats/activity-log/index.jsx [hunk 2]
```diff
@@ -264,12 +238,21 @@ class ActivityLog extends Component {
 			return applySiteOffset( moment.utc( ts_utc ) ).format( YEAR_MONTH ) === selectedMonthAndYear;
 		} );
 
+		if ( isEmpty( logsForMonth ) ) {
+			return (
+				<EmptyContent
+					title={ translate( 'No activity for %s', {
```
Comment (a8ci18n) [inline client/my-sites/stats/activity-log/index.jsx]: Hi! I've found a possible matching string that has already been translated **19** times:
`translate( 'Activity for {{period/}}' )` *ES Score: 7*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by folletto** [comment]
screen shot 2017-08-09 at 13 55 38

It might be a bit overwhelming, but let's start here. I'm ok in pushing this live and reviewing later. 👍

**Pull request merged**